modin/experimental/xgboost/test/test_dmatrix.py

```diff
@@ -0,0 +1,174 @@
+
+rng = np.random.RandomState(1994)
+
+
+class TestDMatrix:
+    def test_dmatrix_numpy_init(self):
+        data = np.random.randn(5, 5)
+        dm = xgb.DMatrix(data)
+        md_data = pd.DataFrame(data)
+        md_dm = mxgb.DMatrix(md_data)
+        assert md_dm.num_row() == dm.num_row() == 5
+        assert md_dm.num_col() == dm.num_col() == 5
+
+        data = np.array([[1, 2], [3, 4]])
+        dm = xgb.DMatrix(data)
+        md_data = pd.DataFrame(data)
+        md_dm = mxgb.DMatrix(md_data)
+        assert md_dm.num_row() == dm.num_row() == 2
+        assert md_dm.num_col() == dm.num_col() == 2
+
+        # 0d array
+        with pytest.raises(ValueError):
+            xgb.DMatrix(np.array(1))
+        with pytest.raises(ValueError):
+            mxgb.DMatrix(pd.DataFrame(np.array(1)))
+        # 3d array
+        data = np.random.randn(5, 5, 5)
+        with pytest.raises(ValueError):
+            xgb.DMatrix(data)
+        with pytest.raises(ValueError):
+            mxgb.DMatrix(pd.DataFrame(data))
+        # object dtype
+        data = np.array([["a", "b"], ["c", "d"]])
+        with pytest.raises(ValueError):
+            xgb.DMatrix(data)
+        with pytest.raises(ValueError):
+            mxgb.DMatrix(pd.DataFrame(data))
+
+    def test_feature_names_slice(self):
+        data = np.random.randn(5, 5)
+
+        # different length
+        with pytest.raises(ValueError):
+            xgb.DMatrix(data, feature_names=list("abcdef"))
+        with pytest.raises(ValueError):
+            mxgb.DMatrix(pd.DataFrame(data), feature_names=list("abcdef"))
+
+        # contains duplicates
+        with pytest.raises(ValueError):
+            xgb.DMatrix(data, feature_names=["a", "b", "c", "d", "d"])
+        with pytest.raises(ValueError):
+            mxgb.DMatrix(pd.DataFrame(data), feature_names=["a", "b", "c", "d", "d"])
+
+        # contains symbol
+        with pytest.raises(ValueError):
+            xgb.DMatrix(data, feature_names=["a", "b", "c", "d", "e<1"])
+        with pytest.raises(ValueError):
+            mxgb.DMatrix(pd.DataFrame(data), feature_names=["a", "b", "c", "d", "e<1"])
+
+        dm = xgb.DMatrix(data)
+        md_dm = mxgb.DMatrix(pd.DataFrame(data))
+        dm.feature_names = list("abcde")
+        md_dm.feature_names = list("abcde")
+        assert dm.feature_names == md_dm.feature_names == list("abcde")
+
+        dm = xgb.DMatrix(data, feature_types="q")
+        md_dm = mxgb.DMatrix(pd.DataFrame(data), feature_types="q")
+        assert dm.feature_types == md_dm.feature_types == list("qqqqq")
+
+        dm.feature_types = "q"
+        md_dm.feature_types = "q"
+        assert dm.feature_types == md_dm.feature_types == list("qqqqq")
+
+        dm.feature_types = list("qiqiq")
+        md_dm.feature_types = list("qiqiq")
+        assert dm.feature_types == md_dm.feature_types == list("qiqiq")
+
+    def test_feature_names(self):
+        data = np.random.randn(100, 5)
+        target = np.array([0, 1] * 50)
```

Comment:
```suggestion
        label = np.array([0, 1] * 50)
```

I propose to use `data`, `label` names

Reply:
this parameter was rename